dashboard/app/jobs.go

```diff
@@ -566,6 +566,18 @@ func pollCompletedJobs(c context.Context, typ string) ([]*dashapi.BugReport, err
 		if reporting.Config.Type() != typ {
 			continue
 		}
+
+		// If the bug is already known to be fixed, invalid or duplicate, do not report the
+		// bisection results.
+		bug := new(Bug)
+		bugKey := keys[i].Parent()
+		if err := db.Get(c, bugKey, bug); err != nil {
+			return nil, fmt.Errorf("job %v: failed to get bug: %v", extJobID(keys[i]), err)
+		}
+		if len(bug.Commits) != 0 || bug.Status == BugStatusDup || bug.Status == BugStatusInvalid {
```

Comment:
Let's use `bug.Status == BugStatusOpen`. Shorter, includes Fixed and that's the condition we use in findBugsForBisection.
Also, let's move this after the cheaper checks (right before createBugReportForJob). This involved accessing the data base in the loop, which will slowdown everything and can lead to timeouts. Consider, we skip lots of bisections, bug during each periodic poll we will need to also fetch Bug for each of them, just to later discard it for other reasons.
While if we swap the checks, we will immediately discard most by just looking at properties of the Job object.
I think we also should mark them as reported to not redo this on each poll. It seems that just calling jobReported on the job should do, so no need to write new code :)

Reply:
> Also, let's move this after the cheaper checks (right before createBugReportForJob). This involved accessing the data base in the loop, which will slowdown everything and can lead to timeouts. Consider, we skip lots of bisections, bug during each periodic poll we will need to also fetch Bug for each of them, just to later discard it for other reasons.
> While if we swap the checks, we will immediately discard most by just looking at properties of the Job object.
> I think we also should mark them as reported to not redo this on each poll. It seems that just calling jobReported on the job should do, so no need to write new code :)

I see, thanks! I've made this change.

> Let's use `bug.Status == BugStatusOpen`. Shorter, includes Fixed and that's the condition we use in findBugsForBisection.

When making this change the test seems to be failing; perhaps because incomingCommandTx() is not updating bug.Status. Should I make some other dashapi call in the test to signal the bug is fixed?

Reply:
Did you do "bug.Status == BugStatusOpen" or "bug.Status != BugStatusOpen"?

Reply:
When I try the following I get a leftover email(syzbot reporting the bisection results).

```
if bug.Status != BugStatusOpen {
	jobReported(c, extJobID(keys[i]))
	continue
}
```

Reply:
Oh, I see, we still need the "len(bug.Commits) != 0" part. When we have a commit, the bug is still open.
If it does not help, what exactly test is failing and with what message?

Reply:
The failing test is the one I have added in this PR, and the failing message is for leftover email(email with bisection results).

```
	util_test.go:158: ERROR: leftover email: syzbot suspects this bug was fixed by commit:
		
		commit 46e65cb4a0448942ec316b24d60446bbd5cc7827
		Author: Author Kernelov <author@kernel.org>
		Date:   Wed Feb 9 04:05:06 2000 +0000
		
		    kernel: add a fix
		
		bisection log:  https://testapp.appspot.com/x/bisect.txt?x=14800000000000
		start commit:   11111111 kernel_commit_title1
		git tree:       repo1 branch1
		final crash:    https://testapp.appspot.com/x/report.txt?x=16800000000000
		console output: https://testapp.appspot.com/x/log.txt?x=12800000000000
		kernel config:  https://testapp.appspot.com/x/.config?x=4d11162a90e18f28
		dashboard link: https://testapp.appspot.com/bug?extid=de50215ef28ac08248cd
		syz repro:      https://testapp.appspot.com/x/repro.syz?x=16000000000000
		C reproducer:   https://testapp.appspot.com/x/repro.c?x=11000000000000
		
		If the result looks correct, please mark the bug fixed by replying with:
		
		#syz fix: kernel: add a fix
		
		For information about bisection process see: https://goo.gl/tpsmEJ#bisection
	util_test.go:331: jobs_test.go:552: API(reporting_poll_bugs): &dashapi.PollBugsRequest{Type:"test"}
	util_test.go:331: jobs_test.go:552: API(reporting_poll_bugs): REPLY: &dashapi.PollBugsResponse{Reports:[]*dashapi.BugReport(nil)}
FAIL
exit status 1
```

Reply:
Is this with "len(bug.Commits) != 0" part or not?
The test adds commits to the bug, but status is not changed.

Reply:
(I just edited the comment above so hopefully it is clearer now. Does that make sense?)

Reply:
Yes, I would prefer the bug.Status != BugStatusOpen check. It's shorter, includes Fixed status and that's the condition we use in findBugsForBisection.

Reply:
Thanks! I've updated the PR, sorry about the confusion!